module/icp/algs/aes/aes_impl.c

```diff
@@ -1584,34 +1584,21 @@ intel_aes_instructions_present(void)
 	unsigned eax, ebx, ecx, edx;
 	unsigned func, subfunc;
 
-	if (cached_result == -1) { /* first time */
-		/* check for an intel cpu */
-		func = 0;
-		subfunc = 0;
-
-		__asm__ __volatile__(
-		    "cpuid"
-		    : "=a" (eax), "=b" (ebx), "=c" (ecx), "=d" (edx)
-		    : "a"(func), "c"(subfunc));
-
-		if (memcmp((char *)(&ebx), "Genu", 4) == 0 &&
-		    memcmp((char *)(&edx), "ineI", 4) == 0 &&
-		    memcmp((char *)(&ecx), "ntel", 4) == 0) {
-			func = 1;
-			subfunc = 0;
-
-			/* check for aes-ni instruction set */
-			__asm__ __volatile__(
-			    "cpuid"
-			    : "=a" (eax), "=b" (ebx), "=c" (ecx), "=d" (edx)
-			    : "a"(func), "c"(subfunc));
-
-			cached_result = !!(ecx & INTEL_AESNI_FLAG);
```

Comment:
I'd recommend that we update this function to call the CPU feature checking code which was added for vectorized RAIDZ in `include/linux/simd_x86.h`.  It will need to be extended slightly to check for the ` aes-ni ` instruction but then all of this code will be in a single place.

Reply:
I've now addressed this